tests/testing/BaseTestHelperTest.py

```diff
@@ -9,23 +7,41 @@ def test_good_assert_result_equal(self):
         self.assert_result_equal(['a', 'b'], ['a', 'b'])
 
     def test_inequality_assert_result_equal(self):
-        with pytest.raises(AssertionError) as ex:
+        with self.assertRaises(AssertionError) as ex:
             self.assert_result_equal(['a', 'b'], ['a', 'b', None])
-        assert '[\'a\', \'b\'] != [\'a\', \'b\', None]' in str(ex.value)
+        self.assertEqual("Lists differ: ['a', 'b'] != ['a', 'b', None]\n\n"
```

Comment:
these tests now rely on the output of pytest - see note above about capturing the assertion and converting it into a string.

That isnt helpful.

And these test cases are not helpful.

What we need are tests of multiple complicated `Result` objects, and showing that the error message is helpful for real scenarios.

And our messages should not use pytest's messages, because they are dumb -- they are automatic and can not know the context of the situation, or know what the bear debugger needs to see in order to be able to diagnose the problem.